StripePaymentSheet/StripePaymentSheetTests/PaymentSheet/TextFieldElement+CardTest.swift

```diff
@@ -336,6 +337,65 @@ class TextFieldElementCardTest: STPNetworkStubbingTestCase {
         }
     }
 
+    class STPAnalyticsClientTestDelegate: NSObject, STPAnalyticsClientDelegate {
+        let didLogBlock: (([String: Any]) -> Void)
+
+        init(didLogBlock: @escaping ([String: Any]) -> Void) {
+            self.didLogBlock = didLogBlock
+        }
+
+        func analyticsClientDidLog(analyticsClient: STPAnalyticsClient, payload: [String: Any]) {
+            didLogBlock(payload)
+        }
+    }
+
+    func testAlertsOnExpected19DigitCard() {
+        // Set a publishable key for the metadata service
+        STPAPIClient.shared.publishableKey = STPTestingDefaultPublishableKey
+
+        // Set up a CardSectionElement:
+        let cardSection = CardSectionElement(
+            collectName: false,
+            defaultValues: .init(),
+            preferredNetworks: nil,
+            cardBrandChoiceEligible: false,
+            hostedSurface: .paymentSheet,
+            theme: .default,
+            analyticsHelper: ._testValue(),
+            cardBrandFilter: .default
+        )
+        let textFieldElement = cardSection.panElement
+
+        // A 16 digit card number that should be 19 digits:
+        let unionPay19_but_16_digits_entered = "6235510000000002"
+
+        // Cache the UnionPay BIN range first:
+        let allRetrievalsAreComplete = expectation(description: "Fetch BIN Range")
+        (textFieldElement.configuration as! TextFieldElement.PANConfiguration).binController.retrieveBINRanges(forPrefix: unionPay19_but_16_digits_entered) { _ in
+            allRetrievalsAreComplete.fulfill()
+        }
+        // Wait for the fetch to complete
+        waitForExpectations(timeout: 10, handler: nil)
```

Comment:
Since we have two different expectations that should happen at two different times of the test, i think we should change the first to:
```
        wait(for: [allRetrievalsAreComplete], timeout: 10)
```

and the second to:
```
        wait(for: [logExpectation], timeout: 10)
```

Reply:
Agreed!